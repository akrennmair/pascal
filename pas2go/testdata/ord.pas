program test;

type ttt = ( foo, bar, baz );

var x : ttt;
	c : char;

begin
	c := 'A';
	writeln('ord(A) = ', ord(c));
	writeln('ord(a) = ', ord('a'));
	x := foo;
	writeln('ord(x) = ', ord(x));
	writeln('ord(baz) = ', ord(baz))
end.
