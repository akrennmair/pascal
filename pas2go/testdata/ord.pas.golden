package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	type (
		ttt int
	)
	const (
		foo ttt = 0
		bar ttt = 1
		baz ttt = 2
	)

	var (
		x ttt
		c byte
	)
	_ = x
	_ = c

	c = 'A'
	system.Writeln("ord(A) = ", int(c))
	system.Writeln("ord(a) = ", int('a'))
	x = foo
	system.Writeln("ord(x) = ", int(x))
	system.Writeln("ord(baz) = ", int(baz))
}
